tinygrad/engine/schedule.py

```diff
@@ -505,7 +532,8 @@ def create_schedule_with_vars(outs:list[UOp]) -> tuple[list[ScheduleItem], dict[
   cache: dict[UOp, UOp] = {}
   # to_uop is removing (many) of the movement ops
   for u in (big_graph:=UOp.sink(*(to_uop(x, ctx, cache) for x in outs))).src: ctx.realizes[u.buf_uop] = u
-  big_graph = graph_rewrite(big_graph, remove_movement_ops+ops_folding+do_realize, ctx)
+  big_graph = graph_rewrite(big_graph, rewrite_views, ctx)
```

Comment:
The spec will go over the reasoning behind this, but the basic idea is that when we have:
```
CAST(EXPAND(src))
```
in dfs order:
1. src gets added to realizes
2. CAST(EXPAND(src)) -> EXPAND(CAST(src))

how do we judge when not to realize src?
What if src has two children, one is an expand and one is COPY(src).